tools/test-ecosystem/data.mjs

```diff
@@ -0,0 +1,85 @@
+/**
+ * @fileoverview Data utilities for ecosystem tests and updates to data.
+ * @author Josh Goldberg
+ */
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+import chalk from "chalk";
+import util from "node:util";
+import path from "node:path";
+
+//-----------------------------------------------------------------------------
+// Types
+//-----------------------------------------------------------------------------
+
+/**
+ * Settings for how to clone, set up, and test an ecosystem plugin.
+ * @typedef {Object} PluginData
+ * @property {string} commit Hash to check out after cloning the plugin.
+ * @property {string} repository Repository URL to clone the plugin from.
+ */
+
+//-----------------------------------------------------------------------------
+// Constants
+//-----------------------------------------------------------------------------
+
+export const pluginDataFilePath = path.join(
+	import.meta.dirname,
+	"plugins-data.json",
+);
+
+//-----------------------------------------------------------------------------
+// Functions
+//-----------------------------------------------------------------------------
+
+/**
+ *
+ * @param {"test" | "update"} action
+ * @returns {[string, PluginData][]}
+ */
+export async function getPlugins(action) {
+	const { values } = util.parseArgs({
+		options: {
+			plugin: {
+				type: "string",
+				help: `The name of the plugin to ${action}, or 'all' for all plugins`,
```

Comment:
Can't we treat `undefined` as all? So we don't need pass `--plugin=all`.

Reply:
I like it 🙂 and can make this change if nobody objects. +1